%% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,english]{sphinxmanual}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax

\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
\edef\sphinxdqmaybe{\ifdefined\DeclareUnicodeCharacterAsOptional\string"\fi}
  \DeclareUnicodeCharacter{\sphinxdqmaybe00A0}{\nobreakspace}
  \DeclareUnicodeCharacter{\sphinxdqmaybe2500}{\sphinxunichar{2500}}
  \DeclareUnicodeCharacter{\sphinxdqmaybe2502}{\sphinxunichar{2502}}
  \DeclareUnicodeCharacter{\sphinxdqmaybe2514}{\sphinxunichar{2514}}
  \DeclareUnicodeCharacter{\sphinxdqmaybe251C}{\sphinxunichar{251C}}
  \DeclareUnicodeCharacter{\sphinxdqmaybe2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}
\usepackage{times}
\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=\small}
\usepackage{geometry}

% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}

\addto\captionsenglish{\renewcommand{\figurename}{Fig.\@ }}
\makeatletter
\def\fnum@figure{\figurename\thefigure{}}
\makeatother
\addto\captionsenglish{\renewcommand{\tablename}{Table }}
\makeatletter
\def\fnum@table{\tablename\thetable{}}
\makeatother
\addto\captionsenglish{\renewcommand{\literalblockname}{Listing}}

\addto\captionsenglish{\renewcommand{\literalblockcontinuedname}{continued from previous page}}
\addto\captionsenglish{\renewcommand{\literalblockcontinuesname}{continues on next page}}
\addto\captionsenglish{\renewcommand{\sphinxnonalphabeticalgroupname}{Non-alphabetical}}
\addto\captionsenglish{\renewcommand{\sphinxsymbolsname}{Symbols}}
\addto\captionsenglish{\renewcommand{\sphinxnumbersname}{Numbers}}

\addto\extrasenglish{\def\pageautorefname{page}}

\setcounter{tocdepth}{2}



\title{WEoN Documentation}
\date{Jun 27, 2019}
\release{}
\author{Network Biology Lab}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{}
\makeindex
\begin{document}

\pagestyle{empty}
\sphinxmaketitle
\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}


Weighted Epigenomic Network (WEoN) is a \sphinxhref{https://cytoscape.org/}{Cytoscape} app that incorporates a filtering method to determine context-specific gene regulatory networks. The method employs diverse data to filter out regulatory connections between genes. WEoN uses histone modifications through Chromatin Inmunoprecipitation followed by DNA sequencing (ChIP-seq), gene expression through RNA-seq, and chromatin accesibility through DNase-seq data. A serie of heuristic filters removes known and inferred regulations (like Transcription factors and regulatory RNAs) from a Gene Regulatory Network (GRN) considered as a Reference Network (e.g. a GRN with all known connections, regardless development stage or cell type). The corresponding context-specific GRN considers weighted information, and the generated network can be further analyzed within the \sphinxhref{https://cytoscape.org/}{Cytoscape} software.

Please write directly to us (\sphinxhref{mailto:leandro.murgas@mayor.cl}{Leandro}, \sphinxhref{mailto:alberto.martin@umayor.cl}{Alberto}) or post an issue at \sphinxhref{https://github.com/networkbiolab/WEoN/issues}{Issues} if you encounter any problem.


\chapter{Installation}
\label{\detokenize{installation:installation}}\label{\detokenize{installation::doc}}
First, be aware that WEoN runs a PERL script that filter out regulations unlikely occurring from a Reference GRN. Please follow the specific instructions for your platform \sphinxhref{https://www.perl.org/get.html}{here}.

\sphinxstylestrong{Requeriments}
\begin{itemize}
\item {} 
Java: Instructions to install Java depends on your operating system. Windows and MacOS users should download Java 8 from \sphinxhref{https://www.java.com/es/download/manual.jsp}{Download\_Java} and follow the installer instructions. For Unix users, Java could be installed from the repository packages \sphinxtitleref{openjdk-8-jdk} and \sphinxtitleref{openjdk-8-jre} (e.g. \sphinxtitleref{apt-get install openjdk-8-jdk openjdk-8-jre}).

\item {} 
Cytoscape: Download the Cytoscape software from \sphinxhref{https://cytoscape.org/download.html}{Download\_Cytoscape}. The webpage will automatically determine your operating system and prompt a download button.

\item {} 
Perl: Similarly to Java, Windows users should install a Perl interpreter. Please download from \sphinxhref{http://strawberryperl.com/}{Download\_Perl} and follow the instructions. For MacOS and Unix operating systems, Perl can be already installed; if not, the user can install it from the repository.

\end{itemize}

\sphinxstylestrong{Download WEoN}

There are two different ways to obtain WEoN:
\begin{enumerate}
\def\theenumi{\arabic{enumi}}
\def\labelenumi{\theenumi .}
\makeatletter\def\p@enumii{\p@enumi \theenumi .}\makeatother
\item {} 
\sphinxstylestrong{Download from the Figshare repository (With data, Recommended).} WEoN can be downloaded from \sphinxhref{https://figshare.com/articles/WEoN\_install\_zip/7913912}{Figshare repository} along with example data. Then, within Cytoscape, go to: Apps \textgreater{}\textgreater{} App Manager \textgreater{}\textgreater{} Install from File…

\end{enumerate}

\noindent{\hspace*{\fill}\sphinxincludegraphics{{download}.png}\hspace*{\fill}}

\sphinxstyleemphasis{OR}
\begin{enumerate}
\def\theenumi{\arabic{enumi}}
\def\labelenumi{\theenumi .}
\makeatletter\def\p@enumii{\p@enumi \theenumi .}\makeatother
\setcounter{enumi}{1}
\item {} 
\sphinxstylestrong{Download from the Github repository (Without data).} If you are familiar
with git, the \sphinxhref{https://github.com/networkbiolab/WEoN}{Github repository} can be cloned and the respective jar file installed from within Cytoscape: Apps \textgreater{}\textgreater{} App Manager \textgreater{}\textgreater{} Install from File…

\end{enumerate}

\sphinxstylestrong{Installation}

Once you have downloaded and uncompress WEoN, and the requirements are met, you can install WEoN as follow
\begin{itemize}
\item {} 
Windows 10: double click on \sphinxtitleref{install.bat} file and follow instructions. Or manually copy the uncompressed directory with data into any directory (e.g. \sphinxtitleref{C:\textbackslash{}users\textbackslash{}your\_user\textbackslash{}Desktop}) and WEoN.jar into the \sphinxtitleref{C:\textbackslash{}Users\textbackslash{}your\_user\textbackslash{}CytoscapeConfiguration\textbackslash{}3apps\textbackslash{}installed} directory.

\item {} 
Unix (Ubuntu): double click on \sphinxtitleref{install.jar} and follow instructions. Or manually copy the uncompressed directory with data into any directory (e.g \sphinxtitleref{/home/your\_user/Desktop}) and WEoN.jar into \sphinxtitleref{/home/your\_user/CytoscapeConfiguration/3/apps/installed}

\item {} 
MacOS: Copy the uncompressed directory with data into \sphinxtitleref{/Users/your\_user} and WEoN.jar into \sphinxtitleref{/Users/your\_user/CytoscapeConfiguration/3/apps/installed}

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxstylestrong{Downloading from the Cytoscape App Store.} We’ll upload WEoN to the Cytoscape
\sphinxhref{https://apps.cytoscape.org/}{App Store} soon, where it could be downloaded. However, example data could only be obtained from the Figshare repository due to file size limitations.
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}
\sphinxstylestrong{Need Help?}
If you run into any problems with installation, please leave an issue in the
official \sphinxhref{https://github.com/networkbiolab/WEoN}{Github repository}.
\end{sphinxadmonition}


\chapter{Instructions to use WEoN}
\label{\detokenize{use_instructions:instructions-to-use-weon}}\label{\detokenize{use_instructions::doc}}
This document contains the principal instructions in order to use the Cytoscape
app WEoN (Weighted Epigenetic Networks). To open WEoN, go to Cytoscape Apps menu,
then click on WEoN to open it.

WEoN use RNA-Seq data to filter out absent transcription factors from a distance-based reference network in addition with other optional data as DNase, Histone Marks and Methylation. We currently applied the method to \sphinxstyleemphasis{Droshophila melanogaster} due to a specific histone marks code that was experimentally determined, although the code can be modified by the user. The example data can be downloaded from \sphinxhref{https://figshare.com/articles/WEoN\_example\_Data/8330024}{data}.
\begin{enumerate}
\def\theenumi{\arabic{enumi}}
\def\labelenumi{\theenumi .}
\makeatletter\def\p@enumii{\p@enumi \theenumi .}\makeatother
\setcounter{enumi}{-1}
\item {} 
\sphinxstylestrong{Open WEoN}

Within Cytoscape, go to: Apps \textgreater{}\textgreater{} WEoN - Weighted Epigenomic Network. This will display the user interface as shown in the figure.

\item {} 
\sphinxstylestrong{Interface}

The WEoN interface is a simple selector screen that serves as input screen for
the backend Perl scripts, therefore parsing correctly the genomic data and
calling orderly the scripts to filter out unlikely ocurring regulations.

\end{enumerate}

\noindent{\hspace*{\fill}\sphinxincludegraphics{{interface}.png}\hspace*{\fill}}
\begin{enumerate}
\def\theenumi{\arabic{enumi}}
\def\labelenumi{\theenumi .}
\makeatletter\def\p@enumii{\p@enumi \theenumi .}\makeatother
\setcounter{enumi}{1}
\item {} 
\sphinxstylestrong{Required data}

WEoN use RNA-seq data to filter out absent transcription factors and miRNAs
from the Reference GRN. We provide three References,
which were constructed within a differente cutoff of 1500, 2000, and 5000
nucleotides from the Transcription Start Site.

Please select a two-column file separated by tabulations as an \sphinxcode{\sphinxupquote{Expression File}}.
The first column is the gene name while the second is the expression in any
unit, like counts, RPKM, or FPKM. WEoN use an internal dictionary to match gene
names from the Reference Network and the Expression File. Data must be a single
experiment or the average value of the experimental replication.

We provide \sphinxcode{\sphinxupquote{expression\_test.tsv}} as an example \sphinxcode{\sphinxupquote{Expression File}}. Please
click on the corresponding \sphinxcode{\sphinxupquote{Select File}} button and navigated to the containing
folder. Also, provide a path (with write permission) from the \sphinxcode{\sphinxupquote{Select Folder}}
button: click on and navigate.

\end{enumerate}

\begin{sphinxadmonition}{note}{Note:}
The resulting Gene Regulatory Network will be stored at the user selected path
from the \sphinxcode{\sphinxupquote{Select Folder}} button. Although, the GRN will be loaded automatically
when WEoN finish the filtering processes, the user can reuse the GRN.
\end{sphinxadmonition}
\begin{enumerate}
\def\theenumi{\arabic{enumi}}
\def\labelenumi{\theenumi .}
\makeatletter\def\p@enumii{\p@enumi \theenumi .}\makeatother
\setcounter{enumi}{2}
\item {} 
\sphinxstylestrong{Optional data}

\sphinxcode{\sphinxupquote{DNase file}} and \sphinxcode{\sphinxupquote{Methylation file}} are four-columns files separated by
tabulations. Each column correspond, in order, to the chromosome where was mapped
the sequence, the initial coordinate, the ending coordinate, and the score for
the mapped feature. Both files has an associated \sphinxcode{\sphinxupquote{Score}} block which the user
can use a threshold value where all lower scores are dismissed. Default is zero,
meaning all mapped features in the \sphinxcode{\sphinxupquote{DNase}} and \sphinxcode{\sphinxupquote{Methylation}} files will
be used in the filtering process.

The \sphinxcode{\sphinxupquote{Histone Mark Path Files}} allows the introduction of a single file that
determine the absolute path to ChIP-seq experiments for each histone post-
tranlational modification. The file is a four-column text as follow:

\end{enumerate}

\noindent{\hspace*{\fill}\sphinxincludegraphics{{marks}.png}\hspace*{\fill}}
\begin{enumerate}
\def\theenumi{\arabic{enumi}}
\def\labelenumi{\theenumi .}
\makeatletter\def\p@enumii{\p@enumi \theenumi .}\makeatother
\setcounter{enumi}{3}
\item {} 
\sphinxstylestrong{Execute filtering}

After all the data is loaded, please click on \sphinxcode{\sphinxupquote{Run WEoN}}, wait, and
It is important to note that when the job is finished, you need to click on \sphinxcode{\sphinxupquote{Create View}} button to display the time/tissue specific GRN in Cytoscape.

\end{enumerate}

\begin{sphinxadmonition}{note}{Note:}
Feel free to contact directly throught the \sphinxhref{https://github.com/networkbiolab/WEoN}{Github repository}
or to Dr. Alberto Martin’s \sphinxhref{mailto:amartin@umayor.cl}{e-mail}.
\end{sphinxadmonition}


\chapter{Indices and tables}
\label{\detokenize{index:indices-and-tables}}\begin{itemize}
\item {} 
\DUrole{xref,std,std-ref}{genindex}

\item {} 
\DUrole{xref,std,std-ref}{modindex}

\item {} 
\DUrole{xref,std,std-ref}{search}

\end{itemize}



\renewcommand{\indexname}{Index}
\printindex
\end{document}