%% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,english]{sphinxmanual}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax

\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
\edef\sphinxdqmaybe{\ifdefined\DeclareUnicodeCharacterAsOptional\string"\fi}
  \DeclareUnicodeCharacter{\sphinxdqmaybe00A0}{\nobreakspace}
  \DeclareUnicodeCharacter{\sphinxdqmaybe2500}{\sphinxunichar{2500}}
  \DeclareUnicodeCharacter{\sphinxdqmaybe2502}{\sphinxunichar{2502}}
  \DeclareUnicodeCharacter{\sphinxdqmaybe2514}{\sphinxunichar{2514}}
  \DeclareUnicodeCharacter{\sphinxdqmaybe251C}{\sphinxunichar{251C}}
  \DeclareUnicodeCharacter{\sphinxdqmaybe2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}
\usepackage{times}
\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=\small}
\usepackage{geometry}

% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}

\addto\captionsenglish{\renewcommand{\figurename}{Fig.\@ }}
\makeatletter
\def\fnum@figure{\figurename\thefigure{}}
\makeatother
\addto\captionsenglish{\renewcommand{\tablename}{Table }}
\makeatletter
\def\fnum@table{\tablename\thetable{}}
\makeatother
\addto\captionsenglish{\renewcommand{\literalblockname}{Listing}}

\addto\captionsenglish{\renewcommand{\literalblockcontinuedname}{continued from previous page}}
\addto\captionsenglish{\renewcommand{\literalblockcontinuesname}{continues on next page}}
\addto\captionsenglish{\renewcommand{\sphinxnonalphabeticalgroupname}{Non-alphabetical}}
\addto\captionsenglish{\renewcommand{\sphinxsymbolsname}{Symbols}}
\addto\captionsenglish{\renewcommand{\sphinxnumbersname}{Numbers}}

\addto\extrasenglish{\def\pageautorefname{page}}

\setcounter{tocdepth}{2}



\title{WEoN Documentation}
\date{Apr 02, 2019}
\release{}
\author{Rodrigo Santibáñez}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{}
\makeindex
\begin{document}

\pagestyle{empty}
\sphinxmaketitle
\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}


WEoN (Weighted Epigenomic Network) is a \sphinxhref{https://cytoscape.org/}{Cytoscape} app that incorporates a
filtering method to determine specific gene regulatory networks. The method uses
histone modifications through CHIp-on-chip seq, gene expression through
RNA-seq, and chromatin accesibility through DNase-seq to filter out known and
inferred regulations (like Transcription factors and regulatory RNAs) from a
Gene Regulatory Network (GRN) considered as a gold standard (e.g. a GRN with
all known connections, regardless development stage or cell type).
The corresponding specific GRN consider weighted information that can be further
analyzed to validate the network.


\chapter{Installation}
\label{\detokenize{installation:installation}}\label{\detokenize{installation::doc}}
First, be aware that WEoN runs a PERL script that filter out regulations unlikely occurring
from a gold standard GRN. Please follow the specific instructions for your
platform \sphinxhref{https://www.perl.org/get.html}{here}. Also, the app backend is in
transition to python, so please also follow intructions to get python3
\sphinxhref{https://www.python.org/about/gettingstarted/}{here}.

There are two different ways to obtain WEoN:
\begin{enumerate}
\def\theenumi{\arabic{enumi}}
\def\labelenumi{\theenumi .}
\makeatletter\def\p@enumii{\p@enumi \theenumi .}\makeatother
\item {} 
\sphinxstylestrong{Download from the Figshare repository (Recommended).} WEoN can be downloaded
from \sphinxhref{https://figshare.com/articles/WEoN\_install\_zip/7913912}{Figshare repository}
along with example data. Then, within Cytoscape, go to: Apps \textgreater{}\textgreater{} App Manager \textgreater{}\textgreater{} Install from File…

\sphinxstyleemphasis{OR}

\item {} 
\sphinxstylestrong{Download from the Github repository.} If you are familiar
with git, the \sphinxhref{https://github.com/networkbiolab/WEoN}{Github repository} can be cloned
and the respective jar file installed from within Cytoscape: Apps \textgreater{}\textgreater{} App Manager \textgreater{}\textgreater{} Install from File…

\end{enumerate}

Additionally, please run \sphinxcode{\sphinxupquote{script.sh}} or copy the PERL scripts to
\sphinxcode{\sphinxupquote{/home/\$USER/CytoscapeConfiguration/3/apps/installed}}
(*UNIX), while similar paths exist in MacOS and Windows OS.
Please be aware you need a PERL interpreter to execute WEoN backend.

\begin{sphinxadmonition}{note}{Note:}
\sphinxstylestrong{Downloading from the Cytoscape App Store.} We’ll upload WEoN to the Cytoscape
\sphinxhref{https://apps.cytoscape.org/}{App Store} soon, where it could be downloaded. However,
example data could only be obtained from the Figshare repository due to size limitations.
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}
\sphinxstylestrong{Need Help?}
If you run into any problems with installation, please leave an issue in the
official \sphinxhref{https://github.com/networkbiolab/WEoN}{Github repository}.
\end{sphinxadmonition}


\chapter{Instructions to use WEoN}
\label{\detokenize{use_instructions:instructions-to-use-weon}}\label{\detokenize{use_instructions::doc}}
This document contains the principal instructions in order to use the Cytoscape
app WEoN (Weighted Epigenetic Networks). To open WEoN, go to Cytoscape Apps menu,
then click on WEoN to open it.

We currently applied the method to \sphinxstyleemphasis{Droshophila melanogaster} due to a specific
epigenetic code referred to the experimentally determined impact of post-
traslational modifications on histones.
\begin{enumerate}
\def\theenumi{\arabic{enumi}}
\def\labelenumi{\theenumi .}
\makeatletter\def\p@enumii{\p@enumi \theenumi .}\makeatother
\item {} 
\sphinxstylestrong{Interface}

The WEoN interface is a simple selector screen that serves as input screen for
the backend PERL scripts, therefore parsing correctly the genomic data and
calling orderly the scripts to filter out unlikely ocurring regulations.

\end{enumerate}

\noindent{\hspace*{\fill}\sphinxincludegraphics{{interface}.png}\hspace*{\fill}}
\begin{enumerate}
\def\theenumi{\arabic{enumi}}
\def\labelenumi{\theenumi .}
\makeatletter\def\p@enumii{\p@enumi \theenumi .}\makeatother
\setcounter{enumi}{1}
\item {} 
\sphinxstylestrong{Required data}

WEoN use RNA-seq data to filter out absent transcription factors and miRNAs
from the Gold Standard (or Reference Network). We provide three Gold Standards,
which were constructed within a differente cutoff of 1500, 2000, and 5000
nucleotides from the Transcription Start Site.

Please select a two-column file separated by tabulations as an \sphinxcode{\sphinxupquote{Expression File}}.
The first column is the gene name while the second is the expression in any
unit, like counts, RPKM, or FPKM. WEoN use an internal dictionary to match gene
names from the Gold Standard and the Expression File. Data must be a single
experiment or the average value of the experimental replica.

We provide \sphinxcode{\sphinxupquote{expression\_test.tsv}} as an example \sphinxcode{\sphinxupquote{Expression File}}. Please
click on the corresponding \sphinxcode{\sphinxupquote{Select File}} button and navigated to the containing
folder. Also, provide a path (with write permission) from the \sphinxcode{\sphinxupquote{Select Folder}}
button: click on and navigate.

\end{enumerate}

\begin{sphinxadmonition}{note}{Note:}
The resulting Gene Regulatory Network will be stored at the user selected path
from the \sphinxcode{\sphinxupquote{Select Folder}} button. Although, the GRN will be loaded automatically
when WEoN finish the filtering processes, the user can reuse the GRN.
\end{sphinxadmonition}
\begin{enumerate}
\def\theenumi{\arabic{enumi}}
\def\labelenumi{\theenumi .}
\makeatletter\def\p@enumii{\p@enumi \theenumi .}\makeatother
\setcounter{enumi}{2}
\item {} 
\sphinxstylestrong{Optional data}

\sphinxcode{\sphinxupquote{DNase file}} and \sphinxcode{\sphinxupquote{Methylation file}} are four-columns files separated by
tabulations. Each column correspond, in order, to the chromosome where was mapped
the sequence, the initial coordinate, the ending coordinate, and the score for
the mapped feature. Both files has an associated \sphinxcode{\sphinxupquote{Score}} block which the user
can use a threshold value where all lower scores are dismissed. Default is zero,
meaning all mapped features in the \sphinxcode{\sphinxupquote{DNase}} and \sphinxcode{\sphinxupquote{Methylation}} files will
be used in the filtering process.

The \sphinxcode{\sphinxupquote{Histone Mark Path Files}} allows the introduction of a single file that
determine the absolute path to ChIP-seq experiments for each histone post-
tranlational modification. The file is a four-column text as follow:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
mark            state   annotation      location
H3K27me3        \PYGZhy{}               promoter        /absolute/path/to/example\PYGZus{}data/H3K27me3\PYGZus{}0\PYGZhy{}4hr.bed
H3K27ac         +               promoter        ...
H3K36me2        +               promoter        ...
H3K36me3        +               promoter        ...
H3K4me1         +               promoter        ...
H3K4me2         +               promoter        ...
H3K4me3         +               promoter        ...
H3K79me2        +               promoter        ...
H3K9ac          +               promoter        ...
H3K9me2         +               promoter        ...
H3K9me3         +               promoter        ...
H3S10ph         +               promoter        ...
H4K16ac         +               promoter        ...
H4K20me3        \PYGZhy{}               promoter        ...
\end{sphinxVerbatim}

\end{enumerate}

\begin{sphinxadmonition}{note}{Note:}
We will improve the annotation of histone marks associating each mark to an
experimentally validated effect on specific DNA sequences like promoters. For
the current release of WEoN, the 3rd column don’t interfere with the filtering
process.
\end{sphinxadmonition}
\begin{enumerate}
\def\theenumi{\arabic{enumi}}
\def\labelenumi{\theenumi .}
\makeatletter\def\p@enumii{\p@enumi \theenumi .}\makeatother
\setcounter{enumi}{3}
\item {} 
\sphinxstylestrong{Execute filtering}

Click on \sphinxcode{\sphinxupquote{Run WEoN}}, wait, and load the time/tissue specific GRN into Cytoscape
clicking on \sphinxcode{\sphinxupquote{Create View}}.

\end{enumerate}

\begin{sphinxadmonition}{note}{Note:}
Feel free to contact directly throught the \sphinxhref{https://github.com/networkbiolab/WEoN}{Github repository}
or to Dr. Alberto Martin’s \sphinxhref{mailto:amartin@umayor.cl}{e-mail}.
\end{sphinxadmonition}


\chapter{Indices and tables}
\label{\detokenize{index:indices-and-tables}}\begin{itemize}
\item {} 
\DUrole{xref,std,std-ref}{genindex}

\item {} 
\DUrole{xref,std,std-ref}{modindex}

\item {} 
\DUrole{xref,std,std-ref}{search}

\end{itemize}



\renewcommand{\indexname}{Index}
\printindex
\end{document}